integration-tests/azure-storage-blob/src/main/java/io/quarkiverse/azure/storage/blob/it/StorageBlobAsyncResource.java

```diff
@@ -0,0 +1,113 @@
+ * limitations under the License.
+ */
+package io.quarkiverse.azure.storage.blob.it;
+
+import static jakarta.ws.rs.core.Response.Status.CREATED;
+import static jakarta.ws.rs.core.Response.Status.NOT_FOUND;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+
+import com.azure.core.util.BinaryData;
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobServiceAsyncClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.models.BlockBlobItem;
+
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+@Path("/quarkus-azure-storage-blob-async")
+@Produces(MediaType.TEXT_PLAIN)
+@ApplicationScoped
+public class StorageBlobAsyncResource {
+
+    @Inject
+    BlobServiceAsyncClient blobServiceAsyncClient;
+
+    @Path("/{container}/{blobName}")
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    public Uni<Response> uploadBlob(
+            @PathParam("container") String container,
+            @PathParam("blobName") String blobName,
+            String body) {
+        Mono<BlockBlobItem> blockBlobItem = blobServiceAsyncClient.createBlobContainerIfNotExists(container)
+                .map(it -> it.getBlobAsyncClient(blobName))
+                .flatMap(it -> it.upload(BinaryData.fromString(body), true));
+
+        return Uni.createFrom().completionStage(blockBlobItem.toFuture()).map(it -> Response.status(CREATED).build());
+    }
+
+    @Path("/{container}/{blobName}")
+    @HEAD
+    public Uni<Response> blobExists(
+            @PathParam("container") String container,
+            @PathParam("blobName") String blobName) {
+        BlobAsyncClient blobAsyncClient = blobServiceAsyncClient.getBlobContainerAsyncClient(container)
+                .getBlobAsyncClient(blobName);
+
+        return Uni.createFrom().completionStage(blobAsyncClient.exists().map(it -> {
+            if (Boolean.TRUE.equals(it)) {
+                return Response.ok().build();
+            } else {
+                return Response.status(NOT_FOUND).build();
+            }
+        }).toFuture());
+    }
+
+    @Path("/{container}/{blobName}")
+    @GET
+    public Uni<Response> downloadBlob(
+            @PathParam("container") String container,
+            @PathParam("blobName") String blobName) {
+        BlobAsyncClient blobAsyncClient = blobServiceAsyncClient.getBlobContainerAsyncClient(container)
+                .getBlobAsyncClient(blobName);
+
+        return Uni.createFrom()
+                .completionStage(blobAsyncClient.downloadContent().map(it -> Response.ok().entity(it.toString()).build())
+                        .toFuture());
+    }
+
+    @Path("/{container}/{blobName}")
+    @DELETE
+    public Uni<Response> delete(
```

Comment:
all other methods seems follow pattern like *blob*;
Eg: uploadBlob, downloadBlob
so, if you follow the pattern, I think `deleteBlob` would be better than `delete`

Reply:
Makes sense, will update later.

Reply:
Fixed in [7c766dc](https://github.com/quarkiverse/quarkus-azure-services/pull/222/commits/7c766dcab0f4ae503373d30b3c2485686c5b8362).